### DISCUSSION TITLE
some questions about uct_iface_attr.cap.am

### DISCUSSION THREAD

**Discussion by w7658610** [openucx/ucx#7803] [Q&A] [answered]
uct_iface_attr.cap.am
max_short,max_bcopy,max_zcopy, has different sizes。
when I use rc_verbs/mlx5_0:1 :
    short : max supported: 123
    bcopy  :  max supported: 8255
    zcopy : ax supported: 8255

when I use tcp/ib0:
    short : max supported: 8192
    bcopy : max supported: 8192
    zcopy : max supported: 65536

Are these parameters unchangeable?
Do I have to shard data if I want to send larger data? thx.

**Comment by brminich** [comment, marked as answer]
These parameters can not be changed during iface lifetime, but you have to query them for every opened iface.
Yes, you'd need to send multiple fragments if you have to send data larger than these thresholds

**Reply by w7658610** [reply]
Isn't ZCopy optimized for large packages?
I am wondering whether 64K is too small？ For RDMA

**Reply by w7658610** [reply]
![image](https://user-images.githubusercontent.com/37858414/146156862-be95869d-dfac-45ca-8f50-5d0ac6649191.png)
![image](https://user-images.githubusercontent.com/37858414/146156965-4c4e9dad-1b78-4e27-98a2-dac6a2bf869e.png)
In uct_hello_world.c , 
when I use uct_iface_query,the max supported became very small。Why is that?
